copycat/api/src/main/java/org/apache/kafka/copycat/sink/SinkTaskContext.java

```diff
@@ -59,6 +63,40 @@ public void offset(Map<TopicPartition, Long> offsets) {
     }
 
     /**
+     * Set the backoff timeout. SinkTasks should use this to indicate that they need to retry certain
```

Comment:
Wait, that seems like the exact opposite of what we want. This method allows tasks to indicate to the framework the maximum amount of time that should be allowed to pass before their `put()` method is invoked again, regardless of whether there are new records available. The first half of @gwenshap's explanation makes sense, but the second half isn't right -- the idea is to use this to let the task inform the framework about reasonable times, not the other way around.

@Ishiihara This is also why I was suggesting we might want to be very clear about separating the interface for `SinkTaskContext` from the implementation. The interface should only support this method `void backoff(long backoffMs)` to allow tasks to set the maximum delay they will see before the next `put()` call. Then the implementation can include `long backoff()`, which is how the framework gets access to that value.